### PR TITLE
[DiskRP] Swagger changes for API version 2019-11-01

### PR DESCRIPTION
**Pull request by realschwa**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/compute/resource-manager/readme.md
Comment (ArcturusZhang) [inline specification/compute/resource-manager/readme.md]: Why we append a disk suffix to this tag? To me, this tag works as a fully functional compute service

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-11-01/disk.json
Comment (majastrz) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2019-11-01/disk.json]: >2019-07-01 [](start = 16, length = 10)

This should be `2019-11-01` to match the directory name.

Comment (majastrz) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2019-11-01/disk.json]: All the examples as well.

---
In reply to: [369914444](https://github.com/Azure/azure-rest-api-specs/pull/8195#discussion_r369914444) [](ancestors = 369914444)

**Review by majastrz — CHANGES_REQUESTED** [review]
The API versions need to match the folder structure. Once that's fixed, I can sign off from ARM side.

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-11-01/disk.json
Comment (pollock4595) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2019-11-01/disk.json]: Sorry, completely forgot to change these

Comment (majastrz) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2019-11-01/disk.json]: np

---
In reply to: [369917621](https://github.com/Azure/azure-rest-api-specs/pull/8195#discussion_r369917621) [](ancestors = 369917621)

**Review by majastrz — APPROVED** [review]
Signed off from ARM side.

**Comment by pollock4595** [comment]
Please do not merge these changes yet since we have not finished deploying the necessary changes in DiskRP to all regions

**Comment by ArcturusZhang** [comment]
Hi @raych1 could you please help us about the LintDiff error?
```
Property named: 'diskIOPSReadOnly', for definition: 'DiskProperties' must follow camelCase style. Example: 'diskIopsReadOnly'
```
Here `IOPS` I believe this is an abbreviation which should be all capitalized, how could we by-pass this?

**Comment by raych1** [comment]
> Hi @raych1 could you please help us about the LintDiff error?
> Property named: 'diskIOPSReadOnly', for definition: 'DiskProperties' must follow camelCase style. Example: 'diskIopsReadOnly'
> 
> Here IOPS I believe this is an abbreviation which should be all capitalized, how could we by-pass this?

@pollock4595, @ArcturusZhang , please suppress this [rule](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/openapi-authoring-automated-guidelines.md#r3016-definitionspropertiesnamescamelcase) if necessary. /CC @jianyexi

**Review by ArcturusZhang — APPROVED** [review]

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-11-01/disk.json
Comment (hyonholee) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2019-11-01/disk.json]: So, 201 is no more returned?  (In previous version, 201 (Created) was in responses.)

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-11-01/disk.json
Comment (hyonholee) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2019-11-01/disk.json]: Type is no more required?

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-11-01/disk.json
Comment (pollock4595) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2019-11-01/disk.json]: Cosmin told me to remove it

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-11-01/disk.json
Comment (pollock4595) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2019-11-01/disk.json]: No -- users can leave it out and we'll infer it from the type of the disk encryption set itself when we get the disk encryption swt itself with the desId. No desId defaults to  in the back EncryptionAtRestWithPlatformKey

**Comment by pollock4595** [comment]
@raych1 I added a directive to suppress the IOPS lint errors mentioned above, but they do not seem to be working. Can you take a look to see if I did this correctly in the readme? Thanks!

**Comment by pollock4595** [comment]
> > Hi @raych1 could you please help us about the LintDiff error?
> > Property named: 'diskIOPSReadOnly', for definition: 'DiskProperties' must follow camelCase style. Example: 'diskIopsReadOnly'
> > Here IOPS I believe this is an abbreviation which should be all capitalized, how could we by-pass this?
> 
> @pollock4595, @ArcturusZhang , please suppress this [rule](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/openapi-authoring-automated-guidelines.md#r3016-definitionspropertiesnamescamelcase) if necessary. /CC @jianyexi

@raych1 I added a directive to suppress the IOPS lint errors mentioned above, but they do not seem to be working. Can you take a look to see if I did this correctly in the readme? Thanks!

**Comment by raych1** [comment]
> Hi @raych1 could you please help us about the LintDiff error?
> Property named: 'diskIOPSReadOnly', for definition: 'DiskProperties' must follow camelCase style. Example: 'diskIopsReadOnly'
> Here IOPS I believe this is an abbreviation which should be all capitalized, how could we by-pass this?
> 
> @pollock4595, @ArcturusZhang , please suppress this rule if necessary. /CC @jianyexi
> 
> @raych1 I added a directive to suppress the IOPS lint errors mentioned above, but they do not seem to be working. Can you take a look to see if I did this correctly in the readme? Thanks!

@pollock4595 , looks like there's typo for `suppress` in your commit. If the suppression is in same file, it can have multiple items in `where` array. Please refer to [this example](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/datafactory/resource-manager/readme.md).

**Comment by pollock4595** [comment]
> > Hi @raych1 could you please help us about the LintDiff error?
> > Property named: 'diskIOPSReadOnly', for definition: 'DiskProperties' must follow camelCase style. Example: 'diskIopsReadOnly'
> > Here IOPS I believe this is an abbreviation which should be all capitalized, how could we by-pass this?
> > @pollock4595, @ArcturusZhang , please suppress this rule if necessary. /CC @jianyexi
> > @raych1 I added a directive to suppress the IOPS lint errors mentioned above, but they do not seem to be working. Can you take a look to see if I did this correctly in the readme? Thanks!
> 
> @pollock4595 , looks like there's typo for `suppress` in your commit. If the suppression is in same file, it can have multiple items in `where` array. Please refer to [this example](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/datafactory/resource-manager/readme.md).

@raych1 Sorry I didn't notice that, thanks!

**Comment by ArcturusZhang** [comment]
Hi @pollock4595 there are conflicts in this PR, after you resolve these, this PR is good to merge

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-11-01/disk.json
Comment (vikramd-ms) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2019-11-01/disk.json]: Add: "ARM id of the shared galley image version from which to create a disk"

Comment (pollock4595) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2019-11-01/disk.json]: added

Comment (pollock4595) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2019-11-01/disk.json]: I'll add 'Required if creating from a Gallery Image. The id of the ImageDiskReference will be the ARM id of the shared galley image version from which to create a disk.' since this object isn't a URI or string,

**Comment by pollock4595** [comment]
> Hi @pollock4595 there are conflicts in this PR, after you resolve these, this PR is good to merge


@ArcturusZhang This is good to merge when you get a chance. Sorry for the delay!

**Pull request merged**
